MaterialLib/MPL/Properties/PermeabilityMohrCoulombFailureIndexModel.cpp

```diff
@@ -0,0 +1,156 @@
+{
+    name_ = std::move(name);
+}
+
+template <int DisplacementDim>
+void PermeabilityMohrCoulombFailureIndexModel<DisplacementDim>::checkScale()
+    const
+{
+    if (!std::holds_alternative<Medium*>(scale_))
+    {
+        OGS_FATAL(
+            "The property 'PermeabilityMohrCoulombFailureIndexModel' is "
+            "implemented on the 'medium' scale only.");
+    }
+}
+
+template <int DisplacementDim>
+PropertyDataType
+PermeabilityMohrCoulombFailureIndexModel<DisplacementDim>::value(
+    VariableArray const& variable_array,
+    ParameterLib::SpatialPosition const& pos, double const t,
+    double const /*dt*/) const
+{
+    auto const& stress_vector = std::get<SymmetricTensor<DisplacementDim>>(
+        variable_array[static_cast<int>(Variable::stress)]);
+
+    auto const& stress_tensor =
+        formEigenTensor<3>(static_cast<PropertyDataType>(stress_vector));
+
+    Eigen::SelfAdjointEigenSolver<Eigen::Matrix<double, 3, 3>>
+        eigenvalue_solver(stress_tensor);
+
+    // Principle stress
+    auto const sigma = eigenvalue_solver.eigenvalues();
+
+    auto k_data = k0_(t, pos);
+
+    double const max_sigma = std::max(std::fabs(sigma[0]), std::fabs(sigma[2]));
+
+    if (max_sigma < std::numeric_limits<double>::epsilon())
+    {
+        return fromVector(k_data);
+    }
+
+    double const sigma_m = 0.5 * (sigma[2] + sigma[0]);
+
+    double const tau_m = 0.5 * std::fabs(sigma[2] - sigma[0]);
+    double f = 0.0;
+    if (sigma_m > t_sigma_max_)
+    {
+        // tensile failure criterion
+        f = sigma_m / t_sigma_max_;
+
+        double const tau_tt =
+            c_ * std::cos(phi_) - t_sigma_max_ * std::sin(phi_);
+
+        if (std::fabs(tau_tt) > std::numeric_limits<double>::epsilon())
+        {
+            f = std::max(f, tau_m / tau_tt);
+        }
+    }
+    else
+    {
+        // Mohr Coulomb failure criterion
+        f = tau_m / (c_ * std::cos(phi_) - sigma_m * std::sin(phi_));
+    }
+
+    if (f >= 1.0)
+    {
+        const double exp_value = std::exp(b_ * f);
+        for (auto& k_i : k_data)
+        {
+            k_i = std::min(k_i + kr_ * exp_value, k_max_);
+        }
+    }
+
+    // Local coordinate transformation is only applied for the case that the
+    // initial intrinsic permeability is given with orthotropic assumption.
+    if (local_coordinate_system_ && (k_data.size() == DisplacementDim))
+    {
+        Eigen::Matrix<double, DisplacementDim, DisplacementDim> const e =
+            local_coordinate_system_->transformation<DisplacementDim>(pos);
+        Eigen::Matrix<double, DisplacementDim, DisplacementDim> k =
+            Eigen::Matrix<double, DisplacementDim, DisplacementDim>::Zero();
+
+        for (int i = 0; i < DisplacementDim; ++i)
+        {
+            Eigen::Matrix<double, DisplacementDim, DisplacementDim> const
+                ei_otimes_ei = e.col(i) * e.col(i).transpose();
+
+            k += k_data[i] * ei_otimes_ei;
+        }
```

Comment:
Rotation should be mentioned in the documentation. That k can be a tensor does not imply rotations.

Reply:
Please see the reply to you last comment. 

If user inputs two or three numbers for the parameter, he knows that a rotation is possible as other permeability or conductivity input (this should be documented in the related rotation classes). If a user inputs one, four or nine numbers for the parameter ( i.e a full tensor), he definitely knows that there will be no rotation needed. This is the consideration about  the rotation in this implementation.

Reply:
Though I still think the rotations should be mentioned in the documentation it is a minor issue and we will see if the users rise more question, in which case additional documentation will be needed.

Reply:
> Though I still think the rotations should be mentioned in the documentation it is a minor issue and we will see if the users rise more question, in which case additional documentation will be needed.

Is the added sentence in the class documentation about the rotation adequate?

Reply:
This is question to the users.

In my opinion it is not describing the full picture. Especially the order of limiting to k_max and rotations is not mentioned, what influence the presence of `local_coordinate_system` has on the result.